apps/studio/src/pages/sites/[siteId]/pages/[pageId]/settings.tsx

```diff
@@ -99,39 +117,48 @@ const PageSettings = () => {
   const toast = useToast({ duration: THREE_SECONDS_IN_MS, isClosable: true })
   const utils = trpc.useUtils()
 
-  const updatePageSettingsMutation = trpc.page.updateSettings.useMutation({
-    onSuccess: async () => {
-      // TODO: we should use a specialised query for this rather than the general one that retrives the page and the blob
-      await utils.page.invalidate()
-      await utils.resource.invalidate()
-      await utils.folder.invalidate()
-      if (toast.isActive(SUCCESS_TOAST_ID)) {
-        toast.close(SUCCESS_TOAST_ID)
-      }
-      toast({
-        id: SUCCESS_TOAST_ID,
-        title: "Saved page settings",
-        description: "Publish this page for your changes to go live.",
-        status: "success",
-      })
-    },
-    onError: (error) => {
-      toast({
-        title: "Failed to save page settings",
-        description: error.message,
-        status: "error",
-      })
-      reset()
-    },
-  })
+  const { mutateAsync: updatePageSettings } =
+    trpc.page.updateSettings.useMutation({
+      onSuccess: async () => {
+        // TODO: we should use a specialised query for this rather than the general one that retrives the page and the blob
+        await utils.page.invalidate()
+        await utils.resource.invalidate()
+        await utils.folder.invalidate()
+        if (toast.isActive(SUCCESS_TOAST_ID)) {
+          toast.close(SUCCESS_TOAST_ID)
+        }
+        toast({
+          id: SUCCESS_TOAST_ID,
+          title: "Saved page settings",
+          description: "Publish this page for your changes to go live.",
+          status: "success",
+        })
+      },
+      onError: (error) => {
+        toast({
+          title: "Failed to save page settings",
+          description: error.message,
+          status: "error",
+        })
+        reset()
+      },
+    })
 
-  const onSubmit = handleSubmit((values) => {
+  const onSubmit = handleSubmit(async ({ meta, ...rest }) => {
     if (isDirty) {
-      updatePageSettingsMutation.mutate(
-        { pageId, siteId, ...values },
-        {
-          onSuccess: () => reset(values),
-        },
+      await updatePageBlob({
+        pageId,
+        siteId,
+        content: JSON.stringify({
+          ...content,
+          meta,
+        }),
+      }).then(() =>
+        updatePageSettings({
+          pageId,
+          siteId,
+          ...rest,
+        }),
```

Comment:
i strongly suggest updating updatePageSettings to take in the `metadata` prop and doing the update in the backend instead of making two network calls. Less chance of possible conflict and overriding stuff too. Don't really need ALL the content.

Reply:
Ah oops, right. Updated in 498d80ea6a377097367350f7ee5ed36b427a5ed6.